static/app/views/dashboardsV2/widgetBuilder/buildSteps/sortByStep/index.tsx

```diff
@@ -1,45 +1,79 @@
+import styled from '@emotion/styled';
+
 import {generateOrderOptions} from 'sentry/components/dashboards/widgetQueriesForm';
 import Field from 'sentry/components/forms/field';
 import SelectControl from 'sentry/components/forms/selectControl';
-import {t} from 'sentry/locale';
+import {t, tn} from 'sentry/locale';
+import space from 'sentry/styles/space';
 import {Organization, SelectValue} from 'sentry/types';
-import {WidgetQuery, WidgetType} from 'sentry/views/dashboardsV2/types';
+import {DisplayType, WidgetQuery, WidgetType} from 'sentry/views/dashboardsV2/types';
 import {generateIssueWidgetOrderOptions} from 'sentry/views/dashboardsV2/widgetBuilder/issueWidget/utils';
 import {IssueSortOptions} from 'sentry/views/issueList/utils';
 
-import {DataSet, SortDirection} from '../../utils';
+import {DataSet, RESULTS_LIMIT, SortDirection} from '../../utils';
 import {BuildStep} from '../buildStep';
 
 import {SortBySelectors} from './sortBySelectors';
 
 interface Props {
   dataSet: DataSet;
+  displayType: DisplayType;
+  onLimitChange: (newLimit: number) => void;
   onQueryChange: (queryIndex: number, newQuery: WidgetQuery) => void;
   organization: Organization;
   queries: WidgetQuery[];
   widgetBuilderNewDesign: boolean;
   widgetType: WidgetType;
   error?: string;
+  limit?: number;
 }
 
 export function SortByStep({
+  displayType,
   onQueryChange,
   queries,
   dataSet,
   widgetBuilderNewDesign,
   widgetType,
   organization,
   error,
+  limit,
+  onLimitChange,
 }: Props) {
   const orderBy = queries[0].orderby;
 
-  return (
-    <BuildStep
-      title={t('Sort by a column')}
-      description={t("Choose one of the columns you've created to sort by.")}
-    >
-      <Field inline={false} error={error} flexibleControlStateSize stacked>
-        {widgetBuilderNewDesign ? (
+  if (widgetBuilderNewDesign) {
+    return (
+      <BuildStep
+        title={
+          displayType === DisplayType.TABLE
+            ? t('Sort by a column')
+            : t('Sort by a y-axis')
+        }
+        description={
+          displayType === DisplayType.TABLE
+            ? t("Choose one of the columns you've created to sort by.")
+            : t("Choose one of the y-axis you've created to sort by.")
+        }
+      >
+        <Field inline={false} error={error} flexibleControlStateSize stacked>
+          {limit && (
+            <ResultsLimitSelector
+              name="resultsLimit"
+              menuPlacement="auto"
+              options={[...Array(RESULTS_LIMIT).keys()].map(resultLimit => {
+                const value = resultLimit + 1;
+                return {
+                  label: tn('Limit to %s result', 'Limit to %s results', value),
+                  value,
+                };
+              })}
+              value={limit}
+              onChange={(option: SelectValue<number>) => {
+                onLimitChange(option.value);
+              }}
+            />
+          )}
```

Comment:
Priscila and I haven't had a chance to sync up about this decision, but we'll organize a chat. I figured we can do it separately as an improvement since this work had progressed enough, but was waiting on my group by work